Modules/BenchPress.Azure/Public/Confirm-RoleAssignment.ps1

```diff
@@ -52,9 +52,8 @@ function Confirm-RoleAssignment {
   }
   Process {
     $params = @{
-      ServicePrincipalName = $ServicePrincipalId
-      RoleDefinitionName   = $RoleDefinitionName
-      Scope                = $Scope
```

Comment:
I'll clarify why I used `$Scope` as a parameter, just to then filter by it and why I think you should add it back as a parameter.

`Get-AzRoleAssignment` can return multiple values depending on the scope parameter. For example, if the scope is a resource group, it will return _all_ role assignments for the service principal within that resource group. For example, access to a specific resource.

However, this doesn't mean the service principal has access to the root resource group. This is why I filter to the role assignment with exactly the scope.

With this change it will still have the same functionality, but the command will be more inefficient as now it will get every single role assignment for the service principal before filtering. 

Hope that makes sense!

Reply:
Ahh yes I see what you mean! That makes sense to me, will revert back to having the Scope in the splat. Thanks for clarifying!

Reply:
Reverted back!